src/semantics/UHExp.re

```diff
@@ -20,34 +20,44 @@ let is_Space =
 type skel_t = Skel.t(op);
 
 [@deriving sexp]
-type t =
+type block =
+  | Block(lines, t)
+and lines = list(line)
+and line =
+  | EmptyLine
+  | LetLine(UHPat.t, option(UHTyp.t), block)
+  | ExpLine(t)
+and t =
```

Comment:
Not sure where I fall on this, especially as I think ahead to unifying the different node types under some common node signature. Like ExpLines, blocks are weird in that you can't actually put your cursor on a block, you can only place your cursor within some constituent. Could we punt on this to a future refactor? It would also make my life easier w.r.t. current cursor on delimiters refactor.